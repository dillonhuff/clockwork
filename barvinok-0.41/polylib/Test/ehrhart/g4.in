# Gauss Anticipated memory: dependence 4
#
# lambda = ( 1 1 1 )
#
# sigma(X and Y) = ( 1 0 0 )
#                  ( 0 1 0 )

# P = i0, Q = j0, R = k0
# S = N
# T = a0, U = a1
#  i  j  k  P  Q  R  S  T  U  cte
6 11
0  1  0 -1  0  0  0  0  0  0  -1   # i = k+1
0  0  1 -1  0  0  0  0  0  0  -1   # j = k+1
1 -1  0  0  0  0  1  0  0  0   1   # i <= k0 + 1
1  1  0  0 -1 -1 -1  0  1  1  -1   # i >= i0 + j0 + k0 - a0 - a1 + 1
1  1  0  0  0  0  0  0  0  0  -1   # i>= 1
1 -1  0  0  0  0  0  1  0  0  -1   # i<= N-1

# 6 params
#  P  Q  R  S  T  U cte
# i0 j0 k0  N a0 a1
9 8
1  0  0  0  1  0  0 -1   # N >= 1
0  1  0 -1  0  0  0 -1   # i0 = k0 + 1
0  0  1 -1  0  0  0 -1   # j0 = k0 + 1
1  0  0  1  0  0  0  0   # k0 >= 0
1  0  0 -1  1  0  0 -2   # k0 <= N-2
1  0  0 -1  0  1  0 -1   # a0 >= k0 + 1
1  0  0  0  1 -1  0  0   # a0 <= N
1  0  0 -1  0  0  1 -1   # a1 >= k0 + 1
1  0  0  0  1  0 -1  1   # a1 <= N + 1


i0 j0 k0 N a0 a1
Memory - Gauss Pivot


# Result :
---------------------------------------
Domain :
         3R - T - U + 2 >= 0

          P - R  -1 = 0
         Q - R  -1 = 0
         - R + T  -1 >= 0
         - R + S  -2 >= 0
         S - T  >= 0
         - R + U  -1 >= 0
         S - U + 1 >= 0
          1 >= 0

Vertices :
[ 3R-T-U+4,  3R-T-U+4,  3R-T-U+3 ]
[ R+2,  R+2,  R+1 ]
Ehrhart Polynomial:
( -1 * P + ( -1 * Q + ( 1 * T + ( 1 * U + 1 )
 )
 )
 )
tststs
( -1 * P + ( -1 * Q + ( 1 * T + ( 1 * U + 1 )
 )
 )
 )


#  P  Q  R  S  T  U cte
# i0 j0 k0  N a0 a1
---------------------------------------
Domain :
         -3R + T + U  -2 >= 0
         R  >= 0

         P - R  -1 = 0
         Q - R  -1 = 0
         - R + U  -1 >= 0
         S - T  >= 0
         - R + S  -2 >= 0
         - R + T  -1 >= 0
         S - U + 1 >= 0
          1 >= 0
 
Vertices :
[ 2,  2,  1 ]
 
Ehrhart Polynomial:
( 1 )

