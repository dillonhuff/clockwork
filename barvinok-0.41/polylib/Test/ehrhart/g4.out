---------------------------------------
Domain:
         i0 - k0  -1 = 0
         j0 - k0  -1 = 0
         -3k0 + a0 + a1  -2 >= 0
         - k0 + a0  -1 >= 0
         - k0 + a1  -1 >= 0
         k0  >= 0
         N - a0  >= 0
         N - a1 + 1 >= 0
         - k0 + N  -2 >= 0
          1 >= 0

Vertices:
[ 1,  1,  0 ]
[ k0+1,  k0+1,  k0 ]

Ehrhart Polynomial:
( 1 * k0 + 1 )

---------------------------------------
Domain:
         i0 - k0  -1 = 0
         j0 - k0  -1 = 0
         3k0 - a0 - a1 + 2 >= 0
         - k0 + a0  -1 >= 0
         - k0 + a1  -1 >= 0
         N - a0  >= 0
         N - a1 + 1 >= 0
         - k0 + N  -2 >= 0
          1 >= 0

Vertices:
[ 3k0-a0-a1+3,  3k0-a0-a1+3,  3k0-a0-a1+2 ]
[ k0+1,  k0+1,  k0 ]

Ehrhart Polynomial:
( -2 * k0 + ( 1 * a0 + ( 1 * a1 + -1 )
 )
 )

